src/System.Text.Json/src/System/Text/Json/Serialization/ClassMaterializer.cs

```diff
@@ -2,37 +2,41 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System.Collections;
-using System.Collections.Generic;
 using System.Diagnostics;
 using System.Reflection;
-using System.Text.Json.Serialization.Converters;
 
 namespace System.Text.Json.Serialization
 {
     internal abstract class ClassMaterializer
     {
         public abstract JsonClassInfo.ConstructorDelegate CreateConstructor(Type classType);
 
-        public abstract object ImmutableCreateRange(Type constructingType, Type elementType);
+        public abstract object ImmutableCreateRange(Type constructingType, Type elementType, bool constructingTypeIsDict);
 
-        protected MethodInfo ImmutableCreateRangeMethod(Type constructingType, Type elementType)
+        protected MethodInfo ImmutableCreateRangeMethod(Type constructingType, Type elementType, bool constructingTypeIsDict)
```

Comment:
bool method parameters always make me a little woozy. 

I know this is internal guts of the serializer but there are some other factorings of this that make more sense. For instance, if you need to support parameterizing the `keyType` in the future, you'd refactor this right away.

Consider a factoring a little bit more like this:

```
protected MethodInfo FindImmutableCollectionCreateRange(Type constructingType, Type elementType)
protected MethodInfo FindImmutableDictionaryCreateRange(Type constructingType, Type elementType)

private MethodInfo FindImmutableCreateRange(Type constructingType)
```

The top two just have the call to `.MakeGenericMethod(...)` and the private method has the bulk of the logic. 

Up to you if you think this is an improvement or not.

Reply:
Agreed 👍  Making this change, thanks.